modules/internal/Atomics.chpl

```diff
@@ -467,15 +467,15 @@ module Atomics {
       return ret;
     }
     inline proc add(value:uint(8), order:memory_order = memory_order_seq_cst):void {
-      on this do atomic_fetch_add_explicit_uint_least8_t(_v, value, order);
+      on this do local atomic_fetch_add_explicit_uint_least8_t(_v, value, order);
```

Comment:
> Do you see performance improvements from this?

Do you mean this local statement, the pass checking for primitives within local blocks, or this PR in general?

(The answer to all of these questions is that I have no specific performance information and havn't made any measurements. This PR is removing overhead though so it seems obvious that it will not hurt performance and it seems likely to be significant improvement in some cases. I added these 'local' statements based upon inspection of the generated C code - without them downEndCount's on statement does not use executeOnFast.)

Within optimizeOnClauses, the on body is marked as not fast because of this:

    (CallExpr move
      (SymExpr 'tmp:_ref(atomic_int_least64_t)')
        (SymExpr 'call_tmp:__wide__ref_atomic_int_least64_t'))

I just updated this PR with a fix.